### PR TITLE
docs(core): add example of custom/ejected github workflow

### PR DESCRIPTION
**Pull request by meeroslav**
## Current Behavior
Our docs suggest using `nrwl/ci` as a default GitHub workflow leading to users believe this is the only way and opening issues with requests we are not willing (or cannot, due to reusable workflow limitations) implement.

## Expected Behavior
We should also have an example of a custom workflow, like we do for other CI providers.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: docs/shared/monorepo-ci-github-actions.md [hunk 1]
```diff
@@ -71,4 +71,130 @@ jobs:
...
+          pids=()
+          # list of commands to be run on main has env flag NX_CLOUD_DISTRIBUTED_EXECUTION set to false
+          NX_CLOUD_DISTRIBUTED_EXECUTION=false npx nx-cloud record -- npx nx format:check & pids+=($!)
+
+          # add other parallel commands here
+          npx nx affected -t lint --parallel=3 & pids+=($!)
+          npx nx affected -t test --parallel=3 --configuration=ci & pids+=($!)
+          npx nx affected -t build --parallel=3 & pids+=($!)
+
+          for pid in \${pids[*]}; do
```
Comment (isaacplmann) [inline docs/shared/monorepo-ci-github-actions.md:147]: what is this section doing?

File: docs/shared/monorepo-ci-github-actions.md [hunk 2]
```diff
@@ -71,4 +71,130 @@ jobs:
...
+          ref: ${{ github.event.pull_request.head.sha }}
+          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
+          fetch-depth: 0
+
+      - uses: actions/checkout@v3
+        name: Checkout [Default Branch]
+        if: ${{ github.event_name != 'pull_request' }}
+        with:
+          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
+          fetch-depth: 0
```
Comment (isaacplmann) [inline docs/shared/monorepo-ci-github-actions.md:115]: This is great.  I've gotten stuck on this part multiple times trying to set up CI.

File: docs/shared/monorepo-ci-github-actions.md (hunk 1 above)
Comment (isaacplmann) [reply docs/shared/monorepo-ci-github-actions.md:147]: it would be nice to add a comment here

**Review by isaacplmann — CHANGES_REQUESTED** [review]

File: docs/shared/monorepo-ci-github-actions.md (hunk 1 above)
Comment (meeroslav) [reply docs/shared/monorepo-ci-github-actions.md:153]: Added! Please re-review

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**